Sources/SwiftCliCore/OffsetUtility.swift

```diff
@@ -0,0 +1,87 @@
+import Foundation
+
+public struct RowInfo: Equatable {
+    var row: [String]
+    var rowNumber: Int
+    var bottomLine: String = ""
+    var stitchSymbols: String = ""
+    var width: Int = 0
+    var leftIncDec: Int = 0
+    var rightIncDec: Int = 0
+    // This is about the number of "empty stitches" that should be added on the left to account for future increases
+    var leftOffset: Int = 0
+    lazy var leftOffsetString: String = String(repeating: "  ", count: leftOffset)
+    lazy var offsetBottomLine: String = leftOffsetString + bottomLine
+    lazy var offsetStitchSymbols: String = leftOffsetString + stitchSymbols
+    lazy var totalRow: String = offsetStitchSymbols + offsetBottomLine
+}
+
+public class OffsetUtility {
+    public init() {}
+
+    public func gatherAllMetaData(stitchArray: [[String]]) -> [RowInfo] {
+        var allRowsMetaData = [] as [RowInfo]
+        let rowsNum = stitchArray.count
+        for row in 0..<(rowsNum) {
+            let newRowMetadata = makeRowMetadata(stitchRow: stitchArray[row], rowNumber: row)
+            allRowsMetaData.append(newRowMetadata)
+            if newRowMetadata.leftIncDec != 0 {
+                for prevRow in (0...(row-1)).reversed() {
+                    allRowsMetaData[prevRow].leftOffset += newRowMetadata.leftIncDec
+                }
+            }
+        }
+        return(allRowsMetaData)
+    }
+
+
+    public func makeRowMetadata(stitchRow: [String], rowNumber: Int) -> RowInfo {
+        var rowData = RowInfo(row: stitchRow, rowNumber: rowNumber)
+        rowData.leftIncDec = findLeftChanges(stitchRow: stitchRow)
+        rowData.rightIncDec = findRightChanges(stitchRow: stitchRow)
+        rowData.width = stitchRow.count
+        if rowNumber == 0 {
+            rowData.bottomLine = ChartConstructor().makeBottomRow(width: rowData.width)
+        } else {
+            rowData.bottomLine = ChartConstructor().makeMiddleRow(width: rowData.width,
+                                                                  rDiff: rowData.rightIncDec,
+                                                                  lDiff: rowData.leftIncDec)
+        }
+        rowData.stitchSymbols = ChartConstructor().makeStitchRow(row: rowData.row)
+        return(rowData)
+    }
+
+    public func findCenter(stitchRow: [String]) -> Int {
+        let center = floor(Double(stitchRow.count/2))
+    return(Int(center))
```

Comment:
Indentation.

Reply:
I think you actually get the result you want without all this type conversion if you just do `return(stitchRow.count/2)`.

Reply:
Indeed! It worked!